src/arm/phyboard.c

```diff
@@ -111,55 +113,59 @@ mraa_phyboard_pwm_init_replace(int pin)
         return NULL;
     }
 
-    sprintf(devpath, SYSFS_CLASS_PWM "pwm%u", plat->pins[pin].pwm.pinmap);
+    snprintf(devpath, length, SYSFS_CLASS_PWM "pwm%u", plat->pins[pin].pwm.pinmap);
```

Comment:
The main point in using `snprintf()` here is to make sure we won't overflow `devpath` no matter what. In case of other `snprintf()` statements where you've used hardcoded numbers, that's ok - let's say, because you control the length of the incoming string, though it's better to avoid magic numbers at all - but in this case you have a `define` here, which may be changed to something longer by someone at a later date, causing an overflow.

As `devpath` is defined as having a length of `MAX_SIZE`, the safest and the simplest approach would be to just use that in the `snprintf()` call, just as you can [see throughout the existing code](https://github.com/intel-iot-devkit/mraa/search?utf8=%E2%9C%93&q=snprintf%28)